### PR TITLE
Distrox

### PR DESCRIPTION
**Pull request by doktoric**

### CODE REVIEW

File: common/src/main/java/com/sequenceiq/cloudbreak/util/NullUtil.java [hunk 1]
```diff
@@ -12,4 +13,11 @@ private NullUtil() {
             consumer.accept(value);
         }
     }
+
+    public static <T, R> R ifNotNullF(T value, Function<T, R> consumer) {
```
Comment (gregito) [inline common/src/main/java/com/sequenceiq/cloudbreak/util/NullUtil.java]: F?

**Comment by lajosrodek** [comment]
@doktoric Updated PR title for clarity.

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXClusterToClusterConverter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    private GatewayV1ToGatewayV4Converter gatewayConverter;
+
+    public ClusterV4Request convert(DistroXClusterV1Request source) {
+
+        Workspace workspace = workspaceService.getForCurrentUser();
+
+        ClusterV4Request response = new ClusterV4Request();
+        response.setKerberosName(null);
+        response.setGateway(ifNotNullF(source.getGateway(), gatewayConverter::convert));
+        response.setAmbari(null);
```
Comment (gregito) [inline core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXClusterToClusterConverter.java]: should we set these values explicitly null?

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXScaleV1RequestToStackScaleV4RequestConverter.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Inject
+    private WorkspaceService workspaceService;
+
+    @Override
+    public StackScaleV4Request convert(DistroXScaleV1Request source) {
+        StackScaleV4Request stackScaleV4Request = new StackScaleV4Request();
+        stackScaleV4Request.setDesiredCount(source.getDesiredCount());
+        stackScaleV4Request.setGroup(source.getGroup());
+        Optional<Stack> stack = stackService.findStackByNameAndWorkspaceId(source.getName(), workspaceService.getForCurrentUser().getId());
```
Comment (gregito) [inline core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXScaleV1RequestToStackScaleV4RequestConverter.java]: this whole checking and assigning could be simplicated with a more functional way with this:

`stackService.findStackByNameAndWorkspaceId(source.getName(), workspaceService.getForCurrentUser().getId())
                .ifPresentOrElse(stack1 -> stackScaleV4Request.setStackId(stack1.getId()), 
                        () -> new BadRequestException("DistroX cluster does not exist with name: " + source.getName()));`

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        request.setInstanceGroups(ifNotNullF(source.getInstanceGroups(), instanceGroupConverter::convert));
+        request.setNetwork(ifNotNullF(source.getNetwork(), networkConverter::convert));
+        request.setAws(ifNotNullF(source.getAws(), stackParameterConverter::convert));
+        request.setAzure(ifNotNullF(source.getAzure(), stackParameterConverter::convert));
+
+        request.setPlacement(getPlacement(source.getEnvironment().getName()));
+        request.setInputs(source.getInputs());
+        request.setTags(ifNotNullF(source.getTags(), this::getTags));
+        request.setSharedService(ifNotNullF(source.getSdx(), this::getSharedService));
+        request.setCustomDomain(null);
```
Comment (gregito) [inline core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java]: also these values should be set to null?

**Pull request merged**